### PR TITLE
Support stop changefeed after checkpoint reaches target ts

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix bug in https://github.com/pingcap/ticdc/issues/745

### What is changed and how it works?

- Add a periodic health checker in owner, currently check whether the checkpoint-ts of each changefeed reaches target-ts, if reaches, dispatch an admin job.
- Add a new admin job type, `AdminFinish`
- Add target ts verification before chagnefeed is created.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to update the documentation

### Release note

- Fix the bug that resources are not released when checkpoint-ts of a changefeed reaches target-ts

### CODE REVIEW

File: cdc/owner.go [hunk 1]
```diff
@@ -612,10 +612,30 @@ func (o *Owner) collectChangefeedInfo(ctx context.Context, cid model.ChangeFeedI
...
+		feedState = model.StateFinished
 	}
 	return nil, status, feedState, nil
 }
 
+func (o *Owner) checkClusterHealth(_ context.Context) error {
+	// check whether a changefeed has finished by comparing checkpoint-ts and target-ts
+	for _, cf := range o.changeFeeds {
+		if cf.status.CheckpointTs == cf.info.GetTargetTs() {
+			log.Info("changefeed replication finished", zap.String("changefeed", cf.id), zap.Uint64("ts", cf.status.CheckpointTs))
```
Comment (overvenus) [inline cdc/owner.go:625]:
```suggestion
			log.Info("changefeed replication finished", zap.String("changefeed", cf.id), zap.Uint64("checkpointts", cf.status.CheckpointTs))
```

Comment (leoppro) [reply cdc/owner.go:625]: ping @amyangfei

Comment (amyangfei) [reply cdc/owner.go]: addressed

**Comment by amyangfei** [comment]
PTAL @overvenus

**Pull request merged**